lighthouse-core/test/audits/lazy-third-party-test.js

```diff
@@ -0,0 +1,131 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const LazyThirdParty = require('../../audits/lazy-third-party.js');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
+const createTestTrace = require('../create-test-trace.js');
+
+const pwaTrace = require('../fixtures/traces/progressive-app-m60.json');
+const pwaDevtoolsLog = require('../fixtures/traces/progressive-app-m60.devtools.log.json');
+const noThirdPartyTrace = require('../fixtures/traces/no-tracingstarted-m74.json');
+
+/* eslint-env jest */
+describe('Lazy load third party resources', () => {
+  it('correctly identifies a lazy loadable third party resource', async () => {
+    const artifacts = {
+      devtoolsLogs: {defaultPass: networkRecordsToDevtoolsLog([
+        {
+          url: 'https://example.com',
+          startTime: 0,
+          endTime: 50,
+          transferSize: 2000,
+        },
+        {
+          url: 'https://widget.intercom.io/widget/tx2p130c',
+          startTime: 100,
+          endTime: 150,
+          transferSize: 4000,
+        },
+        {
+          url: 'https://js.intercomcdn.com/frame-modern.bb95039c.js',
+          startTime: 200,
+          endTime: 250,
+          transferSize: 8000,
+        },
+      ])},
+      traces: {defaultPass: createTestTrace({timeOrigin: 0, traceEnd: 2000})},
+      URL: {finalUrl: 'https://example.com'},
+    };
+
+    const settings = {throttlingMethod: 'simulate', throttling: {cpuSlowdownMultiplier: 4}};
+    const results = await LazyThirdParty.audit(artifacts, {computedCache: new Map(), settings});
+
+    expect(results.score).toBe(0);
+    expect(results.displayValue).toBeDisplayString('1 facade alternative available');
+    expect(results.details.items).toEqual([
+      {
+        productName: 'Intercom Widget',
+        facade: {
+          type: 'link',
+          text: 'React Live Chat Loader',
+          url: 'https://github.com/calibreapp/react-live-chat-loader',
+        },
+        transferSize: 12000,
+        blockingTime: 0,
+        subItems: {type: 'subitems', items: [
+          {
+            url: 'https://js.intercomcdn.com/frame-modern.bb95039c.js',
+            mainThreadTime: 0,
+            blockingTime: 0,
+            transferSize: 8000,
+            startTime: 200,
+            endTime: 250,
+          },
+          {
+            url: 'https://widget.intercom.io/widget/tx2p130c',
+            mainThreadTime: 0,
+            blockingTime: 0,
+            transferSize: 4000,
+            startTime: 100,
+            endTime: 150,
+          },
+        ]},
+      },
+    ]);
+  });
```

Comment:
I know this is getting lengthy (maybe there's a way to share generated test data), but it would be great to get a few more tests exercising things like items before the cutoff aren't included in the results, network requests for a product interleaved with requests from other entities, more than one of the same product, more than one product in the same load. Not all of these are necessarily important, and some can be combined, but good to get coverage, especially so future refactors can trust the tests and not worry about e.g. url ordering all over again :)